flatcar-terraform-hetzner/server-configs/server1.yaml

```diff
@@ -0,0 +1,25 @@
+variant: flatcar
+version: 1.0.0
+
+# This is a simple NGINX example.
+# Replace the below with your own config.
+# Refer to https://www.flatcar.org/docs/latest/provisioning/config-transpiler/configuration/ for more information.
+
+systemd:
+  units:
+    - name: nginx.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=NGINX example
+        After=docker.service
+        Requires=docker.service
```

Comment:
I'm not sure what happens if the unit starts and tries to use the Docker CLI before the service / socket is active. There might be a subtle race condition here - though usually, Docker starts very early in the boot process, sometimes this unit might just be faster. No idea what happens then.

Reply:
Thank you for testing! Makes sense to remove then.

Reply:
Re-added the dependency. Good discussion, thanks both -  I learned something!

Reply:
By the way, there is a race if you omit it because this unit and the enabled docker unit would start at the same time and its up to systemd which to run first, or?

Reply:
Yes, but `docker.socket` is part of `sockets.target` and you would have to ensure that it gets started before this unit, or it's a race.